### PR TITLE
Consistent TransactionPriorityId ordering/equality

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- Rust 1.81 may panic if we sort types with ordering that is inconsistent with equality.
- The `TransactionPriorityId` currently does not consider `id` when ordering, but does for equality. 

#### Summary of Changes
- Make `TransactionPriorityId` ordering consistent with equality + tests
- Make `TransactionIdGenerator` generate in reverse order - to explicitly favor earlier arrived packets

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
@ryoqun thank you for bringing this to my attention. Although we do not actually use the stdlibs sorting for these priorty ids, I think it's best we resolve this inconsistency anyway.

File: core/src/banking_stage/transaction_scheduler/transaction_priority_id.rs [hunk 1]
```diff
@@ -20,7 +19,9 @@ impl TransactionPriorityId {
 
 impl Ord for TransactionPriorityId {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        self.priority.cmp(&other.priority)
+        self.priority
+            .cmp(&other.priority)
+            .then_with(|| self.id.cmp(&other.id))
```
Comment (ryoqun) [inline core/src/banking_stage/transaction_scheduler/transaction_priority_id.rs]: nit: can be write as `(self.priority, self.id).cmp((&(other.priority, other.id))`

even more, i think we just can now `#[derive(Ord)]`. :)

Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/transaction_priority_id.rs]: yeah - those would have been equivalent.
Think deriving is best option!

Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/transaction_priority_id.rs]: cd91f84080

**Review by ryoqun — APPROVED** [review]
lgtm

**Pull request merged**